### PR TITLE
Added content gating based on visibility flag

### PR DESCRIPTION
**Pull request by naz**
Hides content based on combinations of posts visibility options: `public`, `members`, `paid` and respective member's status.

@allouis wanted to check on the format of the `subscribtions` attribute. Should we check for some specific value in `subscriptions` or regular check if it's non-falsy value is enough?

### CODE REVIEW

File: core/server/api/canary/utils/serializers/output/utils/members.js [hunk 1]
```diff
@@ -1,45 +1,42 @@
...
     if (labs.isSet('members')) {
+        const hideFormatsData = hideMembersOnlyContent(attrs, frame);
+
+        if (hideFormatsData) {
+            ['plaintext', 'html'].forEach((field) => {
+                attrs[field] = '';
+            });
+        }
+
         // CASE: Members always adds tags, remove if the user didn't originally ask for them
```
Comment (gargol) [inline core/server/api/canary/utils/serializers/output/utils/members.js]: @allouis do we still need this block that handles `tag` includes added just for members?

Comment (allouis) [reply core/server/api/canary/utils/serializers/output/utils/members.js]: Nope! nice catch - there's also an input serialiser somewhere that is adding the `include=tags` stuff - which can probably be removed

File: core/server/api/canary/utils/serializers/output/utils/members.js [hunk 2]
```diff
@@ -1,45 +1,42 @@
...
 
 const forPost = (attrs, frame) => {
-    const hideFormatsData = hideMembersOnlyContent(attrs, frame);
-    if (hideFormatsData) {
-        ['plaintext', 'html'].forEach((field) => {
-            attrs[field] = '';
-        });
-    }
     if (labs.isSet('members')) {
+        const hideFormatsData = hideMembersOnlyContent(attrs, frame);
```
Comment (allouis) [inline core/server/api/canary/utils/serializers/output/utils/members.js]: Moving into this block means that the visibility attribute will only take affect when members is on - that is most likely correct and you just fixed a bug :sweat_smile: Maybe one to check w/ zimo/john tho?

**Comment by allouis** [comment]
Hey @gargol! The `subscriptions` attributes is an array only of the active/valid subscriptions, so just checking that the length is not 0 is good! This looks good :+1:

**Pull request merged**